modules/renter/contractor/editor.go

```diff
@@ -63,7 +63,13 @@ func (he *hostEditor) EndHeight() types.BlockHeight { return he.contract.EndHeig
 
 // Close cleanly terminates the revision loop with the host and closes the
 // connection.
-func (he *hostEditor) Close() error { return he.editor.Close() }
+func (he *hostEditor) Close() error {
+	// release revising lock
+	he.contractor.mu.Lock()
+	delete(he.contractor.revising, he.contract.ID)
+	he.contractor.mu.Unlock()
+	return he.editor.Close()
```

Comment:
yeah, the `proto.Editor` -> `contractor.Editor` chain is unfortunate. The former does all the real work, the latter just tracks the contracts and metrics (and, eventually, adds thread-safety).